compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETileAndFuse.cpp

```diff
@@ -0,0 +1,403 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "iree/compiler/Codegen/Common/Transforms.h"
+#include "iree/compiler/Codegen/Transforms/Transforms.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/TilingInterfaceImpl.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Patterns.h"
+#include "mlir/Dialect/SCF/Transforms/TileUsingInterface.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/Iterators.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-amdaie-tile-and-fuse"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Lowers the computation within the workgroup count region for the ops
+/// that are handled by default.
+static LogicalResult lowerWorkgroupCount(
+    RewriterBase &rewriter, func::FuncOp entryPointFn,
+    ArrayRef<OpFoldResult> workgroupCount, ArrayRef<int64_t> tileSizes,
+    ArrayRef<int64_t> staticLoopRanges, ArrayRef<int64_t> interchange,
+    ArrayRef<unsigned> partitionedLoops, int maxWorkgroupParallelDims) {
+  FailureOr<IREE::HAL::ExecutableExportOp> exportOp =
+      getEntryPoint(entryPointFn);
+  if (failed(exportOp)) {
+    return entryPointFn.emitOpError(
+        "expected function to be entry point function");
+  }
+  Block *body = exportOp->getWorkgroupCountBody();
+  if (!body) {
+    return exportOp->emitOpError("unexpected empty workgroup count region");
+  }
+  SmallVector<Operation *> countOps;
+  for (Operation &op : *body) {
+    if (isa<IREE::Flow::DispatchWorkgroupCountFromSliceOp,
+            IREE::Flow::DispatchWorkgroupCountFromDagRootOp>(&op)) {
+      countOps.push_back(&op);
+    }
+  }
+  if (countOps.empty()) {
+    // If there are no default handled `flow.dispatch.workgroup_count`
+    // operation, do nothing. do nothing.
+    return success();
+  }
+  if (!llvm::hasSingleElement(countOps)) {
+    return exportOp->emitOpError(
+        "unexpected multiple flow.dispatch.workgroup_count_from_dag_root "
+        "operations "
+        "in body");
+  }
+  return TypeSwitch<Operation *, LogicalResult>(countOps[0])
```

Comment:
Instead of this for now you can just replace the operation with all results being value 1 for now.

Reply:
I didn't understand this point.

Can you please elaborate? Also, the intention behind the same.

Reply:
I can add this, @Abhishek-Varma the intention is that the workgroup count lowering is dummy for the aie backend as we dont have this concept in our programming model, so this logic that I blindly lefted from the cpu backend can be greatly simplified.